packages/config/src/layer2s/types/Layer2Url.ts

```diff
@@ -0,0 +1,16 @@
+interface AlchemyProvider {
+  provider: 'alchemy'
+  slug: string
+}
+
+interface JsonRpcProvider {
+  provider: 'jsonRpc'
+  url: string
+}
+
+export type Layer2Rpc = (AlchemyProvider | JsonRpcProvider) & {
+  type: 'rpc'
+  callsPerMinute?: number
```

Comment:
I would prefer those to be repeated inside AlchemyProvider and JsonRpcProvider. Then the type Layer2Rpc could be dissolved. I think it would be clearer if the Layer2Url was a union of interfaces without any type algebra